doc/decisions/0007-feature-toggles-development.md

```diff
@@ -0,0 +1,92 @@
+# 0007. Feature toggles development
+
+| Date | Tags |
+|---|---|
+| 2018-05-25 | development |
+
+
+## Status
+
+Proposed
+
+## Context
+
+This change describes a methodology of development which we believe will help
+us increase the confidence and speed in the development process.
+
+### What problem are we trying to solve
+
+Currently the time between a PR is sent and the code gets merged is between 1 day
+to in some case 4 to 5 days.
+
+This has some disadvantages:
+
+1) This workflows incentivise developers to make large PRs, as building 
+incrementally but having to wait 2 to 3 days for the code to be merged at each
+step is not desirable and results in loss of context.
+
+2) If a bug is introduced it is harder to fix, as the longer it takes to integrate code 
+the more likely the developer has lost familiarity with the code.
+
+3) Constant rebases introduce an overhead for the developer and can introduce
+subtle bugs, as rebased code is not thoroughly reviewed.
+
+4) Code review is only performed once, but during the manual QA step significant
+changes might be added, which are not reviewed anymore.
+
+### How are we going to address it
+
+We believe feature toggles development https://martinfowler.com/articles/feature-toggles.html
+is a step in the right direction.
+
+The way it would work is:
+
+A developer starts working on a new feature.
+Any new code will be scoped behind a feature flag and:
+
+1) no significant changes are made to the existing codebase
+2) no changes are visible to the user when the feature flag is off
```

Comment:
Can we automatically detect this?

Reply:
I'd say no probably, as a single line might have a huge impact, while changing styling in all the files should be consequence less.

Reply:
Right so no changes in source code might be automatically merged for instance?
Edge case for sure :)

Reply:
Right so no changes in source code might be automatically merged for instance?
Edge case for sure :)

Reply:
What would be a change outside source code? Removing an icon will error crash the app, changing json files can do the same, etc.

Reply:
Adding a new ADR? :D

Reply:
> What would be a change outside source code? Removing an icon will error crash the app, changing json files can do the same, etc.

Also as we discussed previously, it's a bit complicated to hide new/updated dependency or, for instance, different version of status-go by flag. But this change can be breaking.